staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -444,44 +446,47 @@ func (g *Cloud) getFoundInstanceByNames(names []string) ([]*gceInstance, error)
 
 	found := map[string]*gceInstance{}
 	remaining := len(names)
+	var escapedInstanceNames []string
 
-	nodeInstancePrefix := g.nodeInstancePrefix
 	for _, name := range names {
 		name = canonicalizeInstanceName(name)
-		if !strings.HasPrefix(name, g.nodeInstancePrefix) {
-			klog.Warningf("Instance %q does not conform to prefix %q, removing filter", name, g.nodeInstancePrefix)
-			nodeInstancePrefix = ""
-		}
+		escapedInstanceNames = append(escapedInstanceNames, regexp.QuoteMeta(name))
 		found[name] = nil
 	}
 
-	for _, zone := range g.managedZones {
-		if remaining == 0 {
-			break
-		}
-		instances, err := g.c.Instances().List(ctx, zone, filter.Regexp("name", nodeInstancePrefix+".*"))
-		if err != nil {
-			return nil, err
-		}
-		for _, inst := range instances {
+	// Max instances requested per call limited to 500, so results fit on 1 page
+	for i := 0; i < len(names); i += 500 {
+		end := int(math.Min(float64(i), float64(len(names))))
+		nameRegex := fmt.Sprintf("^(%s)$", strings.Join(escapedInstanceNames[i:end], "|"))
```

Comment:
I would like to get an input from GCE team about: what is the performance implication of this change on GCE control plane.

Reply:
+1, is this giant regex string faster/slower/same than our original wildcard